concepts/eventing/backend-crd.md

```diff
@@ -0,0 +1,148 @@
+# Design Proposal - Eventing Backend CRD
+
+## Table of Contents
+
+- [Concept](#concept)
+- [Metadata](#metadata)
+- [Spec](#spec)
+- [Validation](#validation)
+- [Status](#status)
+- [Discovery](#discovery)
+
+## Concept
+
+The goal of this proposal is to design the backend CRD in a flexible way to support:
+
+- Creating multiple backends of different types (e.g. `nats`, `beb` ...etc).
+- Creating multiple backends of the same type (e.g. `beb-instace-1`, `beb-instace-2`).
+- Providing backend specific configurations in a generic way.
+- Discovering the supported backends by Kyma Eventing.
+- Discovering backend specific configurations.
+
+```yaml
+apiVersion: eventing.kyma-project.io/v1alpha1
+kind: EventingBackend
+metadata:
+  name: any
+  namespace: any
+  labels:
+    kyma-project.io/eventing: backend
+spec:
+  type: "nats"
+  config:
+  - maxInFlightMessages: 10
+status:
+  backendType: nats
+  conditions:
+  - lastTransitionTime: "2022-05-06T08:35:11Z"
+    reason: Backend ready
+    status: "True"
+    type: Backend Ready
+  - lastTransitionTime: "2022-05-06T08:35:11Z"
+    reason: Publisher proxy deployment ready
+    status: "True"
+    type: Publisher Proxy Ready
+  - lastTransitionTime: "2022-05-06T08:35:11Z"
+    reason: Subscription controller started
+    status: "True"
+    type: Subscription Controller Ready
+  eventingReady: true
```

Comment:
nit: Why not just call it ready ?

Reply:
@nachtmaar could you please create a follow-up ticket for this?

Reply:
Follow-up is here: https://github.com/kyma-project/kyma/issues/14305